src/v/pandaproxy/schema_registry/store.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "likely.h"
+#include "outcome.h"
+#include "pandaproxy/schema_registry/error.h"
+#include "pandaproxy/schema_registry/types.h"
+
+#include <absl/container/btree_map.h>
+#include <absl/container/node_hash_map.h>
+#include <absl/hash/hash.h>
+
+#include <functional>
+#include <iterator>
+#include <system_error>
+
+namespace pandaproxy::schema_registry {
+
+class store {
+public:
+    ///\brief Insert a schema for a given subject.
+    ///
+    /// If the schema is not registered, register it.
+    /// If the subject does not have this schema at any version, register a new
+    /// version.
+    ///
+    /// return the schema_version and schema_id, and whether it's new.
+    std::pair<subject_version_id, bool>
```

Comment:
pairs and tuples are a poor man's struct. I think a simle named struct would read better.